### PR TITLE
Friendship v1

### PR DESCRIPTION
**Pull request by mohamedSabry0**
We implemented the required associations for friendships and the corresponding features.
The features are:
Users are able to send a friendship invitation.
Users can see a button “Invite to friendship” next to the name of the user who is not their friend yet - on both users’ list and single-user page.
Users are able to see pending friendship invitations sent to them from other users.
Usres are able to accept or reject friendships' invitation.

### CODE REVIEW

File: config/routes.rb
Comment (Dandush03) [inline config/routes.rb:15]: ### OPTIONAL
- [ ] **a better approach will be calling  it with  namespace or path, check [this docs](https://guides.rubyonrails.org/routing.html#overriding-the-new-and-edit-segments)**

**Review by Dandush03 — CHANGES_REQUESTED** [review]
# Status: :red_circle: Require Changes :red_circle: 
### Great work so far, you did excellent work :muscle: please fix the issues pointed bellow
- [x] **Tests:** Before you move on, we need to make sure that you've test well your code, please make sure to test your models and all of your controller's function
# Happy Coding! :nerd_face: :computer:

**Review by trekab — CHANGES_REQUESTED** [review]
### Changes Required! :recycle: 
Hi team! Well done working on the suggested changes from the last review. :+1: There is just one thing to attend to before moving to the next milestone:
- [x] Your linter checks are not running on your PR. When I run `rubocop` locally, I get some linter errors from your code. Please fix this.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[Portfolio](https://trekab.github.io/portfolio/) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

File: app/views/users/_user.html.erb
Comment (acushlakoncept) [inline app/views/users/_user.html.erb]: * [x] There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it. **Resources**
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

File: app/views/users/show.html.erb
Comment (acushlakoncept) [inline app/views/users/show.html.erb]: - [x] There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it.

File: app/views/layouts/application.html.erb
Comment (acushlakoncept) [inline app/views/layouts/application.html.erb]: There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it.

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

### Hello Team!, Great job! 👏  

You've done an awesome job so far 👌 Excellent tests. But there are still some changes that are necessary for this project to be approved.

- [x] Kindly make sure your pull request shows the result of the GitHub Actions: 
Screen Shot 2020-11-26 at 1 58 21 AM

- [x] There should be NO application logic in the views nor helpers (no Post.new, Comment.create, current_user.friends.find_by(...)...). Everything connected with data processing should be encapsulated in models and used in controllers. **Check my comments on your codebase to update accordingly**

- [x] Add unwanted files to the `.gitignore` file before pushing your codes, I am seeing **.DS_Store** among your files.

- [x] In your Posts timeline, I should only see posts written by me (logged-in user) and those I have added as friends. Currently, on your app, every user's post appears on logged-in users timeline ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- [x] A little update your readme file will suffice, Title and description to describe what this app does, remove TBA on authors and kindly make them available.

[OPTIONAL] Rephrasing your PR description to tell what you did/achieved would be more appropriate than more of intentions. Example: **I want to be able to send a friendship invitation** - rephrased to **Users are able to send friendship invitation**

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made laudable progress in this milestone.
However, one more change is required before this milestone can be approved. Please check my comment below and fix as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] **As mentioned by the previous reviewer, Files not related to the project should not be committed: they should be added to .gitignore (e.g. .DS_Store)**
- I can see that you've added the file (DS_Store) to the `.gitignore` file. However, the file is still present in your repo. This is because there's one more step to take before the file can be untracked. [This thread](https://stackoverflow.com/questions/1274057/how-to-make-git-forget-about-a-file-that-was-tracked-but-is-now-in-gitignore) on how to make git forget about a file that was tracked earlier might be of help.


* * *
**Please make the required changes and submit another review request when you are done.**

* * *
Best regards,
**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by t-yanick** [comment]
Thanks for the thread. All untracked instanced of .DS_Store were removed

**Review by teekaytech — APPROVED** [review]
# Status: Approved :green_circle: 

### Congratulations team :tada: 
### This milestone has passed all the mandatory requirements. It is therefore approved :heavy_check_mark:

You can now merge to development.
Good luck with your next milestone.

* * * 
**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Pull request merged**
